src/main/java/com/uber/cadence/samples/hello/HelloCancellation.java

```diff
@@ -0,0 +1,145 @@
+  public static class GreetingWorkflowImpl implements HelloCancellation.GreetingWorkflow {
+
+    /**
+     * Activity stub implements activity interface and proxies calls to it to Cadence activity
+     * invocations. Because activities are reentrant, only a single stub can be used for multiple
+     * activity invocations.
+     */
+    private final HelloCancellation.GreetingActivities activities =
+        Workflow.newActivityStub(HelloCancellation.GreetingActivities.class);
+
+    @Override
+    public String getGreeting(String name) throws Exception {
+      try {
+        Workflow.sleep(Duration.ofSeconds(10));
+        return activities.composeGreeting("Hello", name);
+      } catch (
+          CancellationException
+              e) { // This exception is thrown when a cancellation is requested on the current
+                   // workflow
+        /**
+         * Issue a goodbye activity in response to cancellation in a detached scope. This means that
+         * the cancellation issued on the parent workflow will NOT trigger cancelation on child
+         * workflows
+         */
+        Workflow.newDetachedCancellationScope(() -> activities.sayGoodbye(name));
+      }
+      return "Greeting took too long. So we said goodbye.";
+    }
+  }
+
+  static class GreetingActivitiesImpl implements HelloCancellation.GreetingActivities {
+
+    private List<String> invocations = new ArrayList<>();
+
+    @Override
+    public String composeGreeting(String greeting, String name) throws Exception {
+      invocations.add("composeGreeting");
+      return greeting + " " + name + "!";
+    }
+
+    @Override
+    public String sayGoodbye(String name) {
+      invocations.add("sayGoodbye");
+      return "Goodbye " + name + "!";
+    }
+
+    List<String> getInvocations() {
+      return invocations;
+    }
+  }
+
+  public static void main(String[] args) {
+    // Start a worker that hosts both workflow and activity implementations.
+    Worker.Factory factory = new Worker.Factory(DOMAIN);
+    Worker worker = factory.newWorker(TASK_LIST);
+
+    // Workflows are stateful. So you need a type to create instances.
+    worker.registerWorkflowImplementationTypes(GreetingWorkflowImpl.class);
+
+    // A shared instance is used to show activity invocations.
+    GreetingActivitiesImpl activities = new GreetingActivitiesImpl();
+    worker.registerActivitiesImplementations(activities);
+
+    // Start listening to the workflow and activity task lists.
+    factory.start();
+
+    // Start a workflow execution. Usually this is done from another program.
+    WorkflowClient workflowClient = WorkflowClient.newInstance(DOMAIN);
+    WorkflowOptions workflowOptions =
+        new WorkflowOptions.Builder()
+            .setTaskList(HelloCancellation.TASK_LIST)
+            .setExecutionStartToCloseTimeout(Duration.ofDays(30))
+            .build();
+    WorkflowStub client =
```

Comment:
Why untyped?

Reply:
the typed workflow returned from using newWorkflowStub doesn't have a cancel() method on it. Our java tests seem to use untyped when testing for cancellation. Is there an example used typed workflowStubs or how do we do it?

Reply:
Add task to add ability to cancel workflow using strongly typed stub

Reply:
will do